storagedriver/s3/s3.go

```diff
@@ -168,7 +168,7 @@ func New(params DriverParameters) (*Driver, error) {
 		}
 	}
 
-	if _, err := bucket.List("", "", "", 1); err != nil {
+	if _, err := bucket.List(strings.TrimRight(params.RootDirectory, "/"), "", "", 1); err != nil {
```

Comment:
Could you put a comment to explain the purpose of this statement? (permissions check, it sounds like).